invenio_records_lom/services/schemas/metadata.py

```diff
@@ -56,27 +74,77 @@ def _deserialize(self, value, attr, data, **kwargs):
 
         # validate inner keys
         if not isinstance(langstring_inner, dict):
-            raise self.make_error("invalid")
+            raise self.make_error("invalid_inner_type")
         inner_keys = set(langstring_inner)
         if "#text" not in inner_keys:
             raise self.make_error("missing_langstring_key", keys="#text")
-        if "lang" not in inner_keys and self.lang_exists:
+        if "lang" not in inner_keys and self.lang_exists is True:
             raise self.make_error("missing_langstring_key", keys="lang")
-        if "lang" in inner_keys and not self.lang_exists:
+        if "lang" in inner_keys and self.lang_exists is False:
             raise self.make_error("extraneous_lang")
         if not inner_keys <= {"#text", "lang"}:
             raise self.make_error(
                 "extraneous_keys", keys=sorted(inner_keys - {"#text", "lang"})
             )
 
-        text = langstring_inner["#text"]
-        lang = langstring_inner.get("lang", True)
+        text = sanitize_unicode(str(langstring_inner["#text"]))
```

Comment:
minor: why is it necessary to cast to str?

Reply:
not technically necessary
in my manual testing of schemas, where I can actually assign any value to `#text`, the error-message for assigning an int was hard to read
I don't think there even is a way for it to be not a `str` at this point...
can remove this if you want?